ext/reflection/php_reflection.stub.php

```diff
@@ -105,7 +105,7 @@ public function returnsReference(): bool {}
     public function hasReturnType(): bool {}
 
     /** @tentative-return-type */
-    public function getReturnType(): ?ReflectionType {}
+    public function getReturnType(): ReflectionNamedType|ReflectionUnionType|ReflectionIntersectionType|null {}
```

Comment:
Hmmm.. I'm not sure this is the best way to solve your original issue. If we happened to add a new `ReflectionType` instance then people extending `ReflectionFunctionAbstract` would have to adapt their return types accordingly. And it is not even going to be possible in a BC way since the class won't be available in older PHP versions.

Please also note that you have to use ./build/gen_stub.php to generate the C header code which we use.

Reply:
Yeah I don't understand the point of this change. ``ReflectionType`` exists as a generic interface for anything that can return a type, so making the return type concrete is just... weird.

Reply:
I understand the point.
My motivation was to avoid the check for the availability of the `getName()` method, which is provided by the `ReflectionNamedType`. IE:
```php
$reflectionProperty = new \ReflectionProperty($object, 'foo');
$reflectionType = $reflectionProperty->getType();

assert($reflectionType instanceof \ReflectionNamedType);

$type = $reflectionType->getName();
```

But, in view of the current proposal (where `ReflectionUnionType` and `ReflectionUnionType` are possible types), I guess this is not a feasible solution.

Thank you so much for your feedback.